src/connectors/autocomplete/__tests__/connectAutocomplete-test.ts

```diff
@@ -859,11 +859,10 @@ search.addWidgets([
         1
       );
 
-      instantSearchInstance.mainHelper!.hasPendingRequests = () => true;
-      instantSearchInstance._isSearchStalled = true;
-      instantSearchInstance.scheduleRender();
-
-      await wait(0);
+      // this client never resolves, thus search is stalled
+      instantSearchInstance.client.search = () => new Promise(() => {});
```

Comment:
I think this would work and wouldn't require to access the `instantSearchInstance.client.search` internal.

```js
const searchRef = {
  current: () => {
    return Promise.resolve(
      createMultiSearchResponse(createSingleSearchResponse({ hits }))
    );
  },
};

const instantSearchInstance = instantsearch({
  searchClient: createSearchClient({
    search() {
      return searchRef.current();
    },
  }),
  indexName: 'indexName',
});

// ...

searchRef.current = () => new Promise(() => {});
```